src/backend/distributed/shared_library_init.c

```diff
@@ -939,6 +997,33 @@ RegisterCitusConfigVariables(void)
 		GUC_UNIT_MS,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.force_disk_available",
+		gettext_noop("Set the amount of disk space that is available in bytes"),
+		gettext_noop("Available disk is normally read from the worker. This setting"
+					 "changes that and its value will always interpreted as the"
+					 "size available on the worker. If it's -1 the actual size"
+					 "of the worker will be used. "),
+		&ForceDiskAvailableInBytes,
+		-1, -1, INT_MAX,
+		PGC_USERSET,
+		GUC_NO_SHOW_ALL,
+		NULL, NULL, NULL);
+
+	DefineCustomIntVariable(
+		"citus.force_disk_size",
```

Comment:
What is the use case for `force_disk_size`? In other words, why would anyone need this ever?

Reply:
this is used in testing, in reality noone would need this I think. With this we set the "disk size". This is not visible to users

Reply:
Having a GUC and complexity in the code seems can be avoided by changing the function during the test:
```SQL

create or replace function pg_catalog.citus_local_disk_space_stats(OUT  available_disk_size bigint, OUT total_disk_size bigint)

as $BODY$
begin
    select 10000 into available_disk_size;
select 200000 into total_disk_size;

end
$BODY$ language plpgsql;
```

and then replace back to the original definition for the rest of the test.

Is this enough to avoid 2 GUCs and the related code?